dags/pull_miovision.py

```diff
@@ -33,11 +32,62 @@
                  'on_failure_callback': task_fail_slack_alert
                 }
 
-dag = DAG(dag_id = dag_name, default_args=default_args, schedule_interval='0 3 * * *')
+@dag(dag_id=dag_name,
+     default_args=default_args,
+     schedule_interval='0 3 * * *',
+     catchup=False)
+def pull_miovision_dag(): 
 # Add 3 hours to ensure that the data are at least 2 hours old
 
-t1 = BashOperator(
+    #this task group checks if necessary to create new partitions and if so, exexcute.
+    @task_group
+    def check_partitions():
+        YEAR = '{{ macros.ds_format(ds, "%Y-%m-%d", "%Y") }}'
+        MONTH = '{{ macros.ds_format(ds, "%Y-%m-%d", "%m") }}'
+
+        @task.short_circuit(ignore_downstream_trigger_rules=False) #only skip immediately downstream task
+        def check_annual_partition(ds=None): #check if Jan 1 to trigger partition creates. 
+            start_date = datetime.strptime(ds, '%Y-%m-%d')
+            if start_date.month == 1 and start_date.day == 1:
+                return True
+            return False
+      
+        create_annual_partition = PostgresOperator(
+            task_id='create_annual_partitions',
+            sql=["SELECT miovision_api.create_yyyy_volumes_partition('volumes', {{ params.year }}::int, 'datetime_bin')",
+                 "SELECT miovision_api.create_yyyy_volumes_15min_partition('volumes_15min', {{ params.year }}::int)",
+                 "SELECT miovision_api.create_yyyy_volumes_15min_partition('volumes_15min_mvt', {{ params.year }}::int)"],
+            postgres_conn_id='miovision_api_bot',
+            params={"year": YEAR},
+            autocommit=True
+        )
+
+        @task.short_circuit(ignore_downstream_trigger_rules=False) #only skip immediately downstream task
+        def check_month_partition(ds=None): #check if 1st of Month to trigger partition creates. 
+            start_date = datetime.strptime(ds, '%Y-%m-%d')
+            if start_date.day == 1:
+                return True
+            return False
+        
+        create_month_partition = PostgresOperator(
+            task_id='create_month_partition',
+            sql="SELECT miovision_api.create_mm_nested_volumes_partition('volumes', {{ params.year }}::int, {{ params.month }}::int)",
+            postgres_conn_id='miovision_api_bot',
+            params={"year": YEAR,
+                    "month": MONTH},
+            autocommit=True
+        )
+
+        check_annual_partition() >> create_annual_partition
+        check_month_partition() >> create_month_partition
+
+    t1 = BashOperator(
         task_id = 'pull_miovision',
```

Comment:
Can you update the directory to 
`/data/airflow/airflow_venv/bin/python3 /data/airflow/data_scripts/volumes/miovision/api/intersection_tmc.py run-api --path /data/airflow/data_scripts/volumes/miovision/api/config.cfg`